### PR TITLE
6 ➡️ 9

### PR DESCRIPTION
**Pull request by chapulina**
# ➡️ Forward port

Port `sdf6` to `sdf9`

Branch comparison: https://github.com/ignitionrobotics/sdformat/compare/sdf9...sdf6

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

**Review by nkoenig — APPROVED** [review]

File: src/parser.cc
Comment (azeey) [inline src/parser.cc:1371]: In https://github.com/ignitionrobotics/sdformat/pull/597, I see tests related to this in `test/integration/nested_model.cc`. Should we not merge those forward as well?

File: test/integration/frame.cc
Comment (azeey) [inline test/integration/frame.cc:1305]: As discussed in https://github.com/ignitionrobotics/sdformat/pull/596#pullrequestreview-688959663, this test will fail because the way flattening works has changed in `sdf9`. I think this test can be removed though since the functionality is tested elsewhere as part of frame semantics. @scpeters can you confirm?

**Review by azeey — CHANGES_REQUESTED** [review]

File: test/integration/frame.cc
Comment (scpeters) [reply test/integration/frame.cc:1305]: I'm not sure we have a test for multiple levels of nested model names. I'll see if I can update the test logic to use SemanticPose for the expectations

Comment (scpeters) [reply test/integration/frame.cc:1305]: updated to use `SemanticPose` API in https://github.com/ignitionrobotics/sdformat/pull/738/commits/d6fab5984a3b5ae0e8d6a551babad4b4a38c3c2c

**Comment by scpeters** [comment]
I also fixed a compilation error in https://github.com/ignitionrobotics/sdformat/pull/738/commits/331ea5386a373fef562280e98023199c0d635aa3

File: src/parser.cc
Comment (scpeters) [reply src/parser.cc:1371]: https://github.com/ignitionrobotics/sdformat/pull/738/commits/38be09c68fd15e308710717e02f128cca9a1d4da

**Comment by nkoenig** [comment]
Thanks @scpeters and @azeey. CI looks green now.

**Comment by chapulina** [comment]
Thanks for sorting it out, @scpeters ! I'll wait for @azeey 's approval

**Review by azeey — APPROVED** [review]

**Pull request merged**
